CRM/Core/PseudoConstant.php

```diff
@@ -738,22 +738,22 @@ public static function &stateProvince($id = FALSE, $limit = TRUE) {
    *   array reference of all State/Province abbreviations.
    */
   public static function stateProvinceAbbreviation($id = FALSE, $limit = TRUE) {
-    if ($id > 1) {
-      $query = "
-SELECT abbreviation
+    if ($id && is_numeric($id)) {
+      if (!array_key_exists($id, (array) self::$stateProvinceAbbreviation)) {
+        $query = "SELECT abbreviation
 FROM   civicrm_state_province
 WHERE  id = %1";
-      $params = array(
-        1 => array(
-          $id,
-          'Integer',
-        ),
-      );
-      return CRM_Core_DAO::singleValueQuery($query, $params);
+        $params = array(
+          1 => array(
+            $id,
+            'Integer',
+          ),
+        );
+        self::$stateProvinceAbbreviation[$id] = CRM_Core_DAO::singleValueQuery($query, $params);
+      }
+      return self::$stateProvinceAbbreviation[$id];
     }
-
-    if (!self::$stateProvinceAbbreviation || !$id) {
```

Comment:
I should note that removing if (!self::$stateProvinceAbbreviation ) seems a bit odd but
1) the self::populate function actually does a cache call of sorts and
2) there is a risk that it is mis-populated if called previously with $limit - whereas that is managed in the cache call b including the where clause in the key